src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -195,12 +195,20 @@ fsm_init_primary(Keeper *keeper)
 		(!file_exists(pgSetup.ssl.serverKey) ||
 		 !file_exists(pgSetup.ssl.serverCert)))
 	{
-		if (!pg_create_self_signed_cert(&pgSetup, config->nodename))
+		if (!pg_create_self_signed_cert(&(config->pgSetup),
+										config->nodename))
 		{
 			log_error("Failed to create SSL self-signed certificate, "
 					  "see above for details");
 			return false;
 		}
+
+		/* update our configuration with ssl server.{key,cert} */
+		if (!keeper_config_write_file(config))
+		{
+			/* errors have already been logged */
+			return false;
+		}
```

Comment:
This needs to be done in `standby_init_database` too. Otherwise secondary doesn't have the self signed certificates in there.

Reply:
I think the `pg_autoclt` configuration management is done at other places, in `standby_init_database` we are only re-creating the certificates so that at the same PATH as on the primary we find new files, not the same server key as on the primary, that `pg_basebackup` sent over to us...